ddp-workspace/projects/ddp-dsm-ui/src/app/guards/checkAuth.guard.ts

```diff
@@ -0,0 +1,21 @@
+import {Injectable} from '@angular/core';
+import {ActivatedRouteSnapshot, CanActivate, Router, RouterStateSnapshot, UrlTree} from '@angular/router';
+import {Observable} from 'rxjs';
+import {Auth} from '../services/auth.service';
+import {ComponentService} from '../services/component.service';
+
+@Injectable()
+export class CheckAuthGuard implements CanActivate {
+  constructor(private auth: Auth, private router: Router) {
+  }
+
+  canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot):
+    Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
+    const selectedRealm = localStorage.getItem(ComponentService.MENU_SELECTED_REALM);
+    if (!this.auth.authenticated() && !selectedRealm) {return true;}
```

Comment:
personally I always prefer to extract boolean expressions as methods even if they are simple ones, I think this technique provides much better understanding and easy to follow the flow, for example `isNotAuthenticated` and `isAuthenticated`